__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -0,0 +1,1665 @@
+      expect(outputObj).toHaveProperty('script');
+      expect(outputObj).toHaveProperty('decoded');
+      expect(outputObj).toHaveProperty('spent_by', null);
+      expect(outputObj).toHaveProperty('selected_as_input', false);
+      expect(outputObj.decoded).toHaveProperty('type', 'P2PKH');
+
+      if (outputObj.value === txValue) {
+        expect(outputObj.decoded).toHaveProperty('address', txDestinationAddress);
+      } else {
+        expect(outputObj.decoded).toHaveProperty('address', txChangeAddress);
+      }
+    }
+
+    // Validating that the fundTx now has its output spent by moveTx
+    const fundTx = history[fundTxId];
+    for (const output of fundTx.outputs) {
+      // We are only interested on the output of this wallet, not the Genesis change output
+      if (output.decoded.address !== fundDestinationAddress) {
+        continue;
+      }
+
+      expect(output.spent_by).toEqual(moveTx.tx_id);
+    }
+  });
+
+  it('should return full history (custom token)', async () => {
+    const hWallet = await generateWalletHelper();
+    await GenesisWalletHelper.injectFunds(
+      hWallet.getAddressAtIndex(0),
+      10
+    );
+    const { hash: tokenUid } = await createTokenHelper(
+      hWallet,
+      'Full History Token',
+      'FHT',
+      100
+    );
+
+    const history = hWallet.getFullHistory();
+    expect(Object.keys(history)).toHaveLength(2);
+
+    // Validating create token properties ( all others have been validated on the previous test )
+    expect(history).toHaveProperty(tokenUid);
+    const createTx = history[tokenUid];
+
+    // Validating inputs
+    expect(createTx.inputs).toHaveLength(1);
+    const inputObj = createTx.inputs[0];
+    expect(inputObj.token).toEqual(HATHOR_TOKEN_CONFIG.uid);
+    expect(inputObj.token_data).toEqual(TOKEN_DATA.HTR);
+    expect(inputObj.value).toEqual(10);
+
+    // Validating outputs
+    expect(createTx.outputs).toHaveLength(4);
+    const changeOutput = createTx.outputs.find(o => o.value === 9);
+    expect(changeOutput).toBeDefined();
+    expect(changeOutput.token).toEqual(HATHOR_TOKEN_CONFIG.uid);
+    expect(changeOutput.token_data).toEqual(TOKEN_DATA.HTR);
+
+    const tokenOutput = createTx.outputs.find(o => o.value === 100);
+    expect(tokenOutput).toBeDefined();
+    expect(tokenOutput.token).toEqual(tokenUid);
+    expect(tokenOutput.token_data).toEqual(TOKEN_DATA.TOKEN);
+
+    const mintOutput = createTx.outputs.find(o => {
+      const isAuthority = wallet.isAuthorityOutput(o);
+      const isMint = o.value === TOKEN_MINT_MASK;
+      return isAuthority && isMint;
+    });
+    expect(mintOutput).toBeDefined();
+    expect(mintOutput.token).toEqual(tokenUid);
+
+    const meltOutput = createTx.outputs.find(o => {
+      const isAuthority = wallet.isAuthorityOutput(o);
+      const isMelt = o.value === TOKEN_MELT_MASK;
+      return isAuthority && isMelt;
+    });
+    expect(meltOutput).toBeDefined();
+    expect(meltOutput.token).toEqual(tokenUid);
```

Comment:
If the history has the token name/symbol, I feel we should add here as well the `expect`

Reply:
It has. Added validation for them on [c038fe58](https://github.com/HathorNetwork/hathor-explorer-service/pull/174/commits/c038fe586b598aa9e4ab37ce4c41afa656b4fea2).